module/src/sandbox.rs

```diff
@@ -22,13 +38,22 @@ pub trait Sandboxer {
     /// Returns a list of module types that can be loaded by this `Sandboxer`.
     fn supported_module_types(&self) -> &'static [&'static str];
 
-    /// Loads the module identified by the given `hash` into a [`Sandbox`].
+    /// Loads the module in the given `path` into a [`Sandbox`] and pass the given
+    /// `init` and `exports` to the module for initialization.
     ///
     /// The corresponding module must have been imported into the module repository
-    /// configured for the current Foundry host.
+    /// configured for the current Foundry host. That is why this method accepts a `path`
+    /// to identify a module.
+    ///
+    /// The `init` serves as configuration parameters for the module-wide initialization,
+    /// and must be CBOR encoded. The `exports` instruct how to instantiate an ordered list
+    /// of service objects to be exported via links. Each item in the `exports` designates
+    /// a call on a module's constructor function, where the first element is name
+    /// of a constructor function, and the second element is arguments to the constructor
+    /// function encoded in CBOR.
     ///
     /// [`Sandbox`]: ./trait.Sandbox.html
-    fn load(&self, hash: &dyn AsRef<Path>) -> Result<Arc<dyn Sandbox>>;
+    fn load(&self, path: &dyn AsRef<Path>, init: &[u8], exports: &[(&str, &[u8])]) -> Result<Arc<dyn Sandbox>>;
```

Comment:
I didn't understand clearly with this. What is the relationship between this `exports` and that in method `Port::export()`?

Reply:
`exports` are for creating a list of service objects. Once created, all the links may only share some of them. No creation of additional service objects at link creation time.